lib/file-list.js

```diff
@@ -70,20 +70,20 @@ var List = function (patterns, excludes, emitter, preprocess, batchInterval) {
 
   var self = this
 
-  // Emit the `file_list_modified` event.
-  // This function is debounced to the value of `batchInterval`
-  // to avoid spamming the listener.
-  this._emitModified = function () {
-    self._emitter.emit('file_list_modified', self.files)
-
-    self._emitModified = _.throttle(function () {
+    // Emit the `file_list_modified` event.
+    // This function is throttled to the value of `batchInterval`
+    // to avoid spamming the listener.
+  ;(function () {
+    function emit () {
       self._emitter.emit('file_list_modified', self.files)
-    }, self._batchInterval, {
-      leading: false
-    })
-  }
-}
+    }
+    var throttledEmit = _.throttle(emit, self._batchInterval, {leading: false})
+    self._emitModified = function (immediate) {
+      immediate ? emit() : throttledEmit()
+    }
+  }())
```

Comment:
No need to hide these so much, you can just put the outside the constructor like this

```js
function emit (self) {...}
function throttledEmit(self) {...}

...

self._emitModified = function (immediate) {
  immediate ? emit(self) : throttledEmit(self)
}
```